src/steamship/invocable/invocable.py

```diff
@@ -111,43 +114,49 @@ def __init__(self, client: Steamship = None, config: Dict[str, Any] = None):
 
     def __init_subclass__(cls, **kwargs):
         super().__init_subclass__(**kwargs)
+
+        start_time = time.time()
+        cls._package_spec = PackageSpec(name=cls.__name__, doc=cls.__doc__, methods=[])
         cls._method_mappings = defaultdict(dict)
         base_fn_list = [
             may_be_decorated
             for base_cls in cls.__bases__
             for may_be_decorated in base_cls.__dict__.values()
         ]
-
         for attribute in base_fn_list + list(cls.__dict__.values()):
             decorator = getattr(attribute, "decorator", None)
             if decorator:
                 if getattr(decorator, "__is_endpoint__", False):
                     path = getattr(attribute, "__path__", None)
                     verb = getattr(attribute, "__verb__", None)
-                    cls._register_mapping(name=attribute.__name__, verb=verb, path=path)
-
-    @staticmethod
-    def _clean_path(path: str = "") -> str:
-        if not path:
-            path = "/"
-        elif path[0] != "/":
-            path = f"/{path}"
-        return path
+                    method_spec = cls._register_mapping(
+                        name=attribute.__name__, verb=verb, path=path
+                    )
+                    cls._package_spec.methods.append(method_spec)
+
+        # Add the HTTP GET /__dir__ method which returns a serialization of the PackageSpec.
+        # Wired up to both GET and POST for convenience (since POST is the default from the Python client, but
+        # GET is the default if using from a browser).
+        cls._register_mapping(name="__steamship_dir__", verb=Verb.GET, path="/__dir__")
+        cls._register_mapping(name="__steamship_dir__", verb=Verb.POST, path="/__dir__")
+        end_time = time.time()
+        logging.info(f"Registered package functions in {end_time - start_time} seconds.")
```

Comment:
How long does this generally take?

Reply:
I don't know; that's why I added it.

This adds a bit of Python introspection to the Package "bootup" process, which happens on every single invocation. I'm not sure how must that costs so figured we could keep track.

An alternative implementation would be to only generate this introspective manifest when __dir__ is actually called. TBH that's probably the better approach.. just didn't occur to me until I'd already written all the code.